triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/TriggerRunner.scala

```diff
@@ -35,14 +39,27 @@ class TriggerRunner(
 
   import TriggerRunner.{Message, Stop}
 
+  // Spawn a trigger runner impl. Supervise it. If it fails to start
+  // its runner, send it a stop message (the server will later send us
+  // a stop message in due course in this case so this actor will get
+  // garbage collected too). If it something bad happens when the
+  // trigger is running, try to restart it up to 3 times.
   private val child =
-    ctx.spawn(Behaviors.supervise(TriggerRunnerImpl(config)).onFailure(restart), name)
+    ctx.spawn(
+      Behaviors
+        .supervise(
+          Behaviors
+            .supervise(TriggerRunnerImpl(ctx.self, config))
+            .onFailure[InitializationException](stop))
+        .onFailure[RuntimeException](
+          restart.withLimit(maxNrOfRetries = 3, withinTimeRange = 10.seconds)),
```

Comment:
hardcoding these kind of values always comes back to bite you. We should make it configurable.

Reply:
Yes. Follow up PR.

Reply:
Never mind, doing it in this one.